auto-value-gson/src/test/java/com/ryanharter/auto/value/gson/AutoValueGsonExtensionTest.java

```diff
@@ -1169,18 +1360,867 @@ public void simple() {
+        + "          case \"c\": {\n"
+        + "            c = Optional.of(cAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"_D\": {\n"
+        + "            d = Optional.of(dAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"e\": {\n"
+        + "            e = Optional.of(eAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"f\": {\n"
+        + "            f = Optional.of(fAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"g\": {\n"
+        + "            g = Optional.of(gAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"h\": {\n"
+        + "            h = Optional.of(hAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"_I_1\":\n"
+        + "          case \"_I_2\":\n"
+        + "          case \"_I\": {\n"
+        + "            i = Optional.of(iAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"j\": {\n"
+        + "            j = Optional.of(jAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"k\": {\n"
+        + "            k = Optional.of(kAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"l\": {\n"
+        + "            l = Optional.of(lAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"m\": {\n"
+        + "            m = Optional.of(mAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"n\": {\n"
+        + "            n = Optional.of(nAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"o\": {\n"
+        + "            o = OptionalInt.of(oAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"p\": {\n"
+        + "            p = OptionalLong.of(pAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          case \"q\": {\n"
+        + "            q = OptionalDouble.of(qAdapter.read(jsonReader));\n"
+        + "            break;\n"
+        + "          }\n"
+        + "          default: {\n"
+        + "            jsonReader.skipValue();\n"
+        + "          }\n"
+        + "        }\n"
+        + "      }\n"
+        + "      jsonReader.endObject();\n"
+        + "      return new AutoValue_Test(a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q);\n"
+        + "    }\n"
```

Comment:
Note that default setters take the wrapped type. Because null can be passed, `ofNullable` is used here.